src/arkode/arkode_io.c

```diff
@@ -34,6 +35,360 @@
+      return retval;
+    }
+    *arg_used = SUNTRUE;
+  }
+  return ARK_SUCCESS;
+}
+
+int ARKodeSetFromCommandLine(void* arkode_mem, const char* arkid, int argc,
+                             char* argv[])
+{
+  ARKodeMem ark_mem;
+  if (arkode_mem == NULL)
+  {
+    arkProcessError(NULL, ARK_MEM_NULL, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NO_MEM);
+    return (ARK_MEM_NULL);
+  }
+  ark_mem = (ARKodeMem)arkode_mem;
+
+  /* Set list of integer command-line arguments, and the corresponding set routine */
+  static struct arkKeyIntPair int_pairs[] =
+    {{"order", ARKodeSetOrder},
+     {"interpolant_degree", ARKodeSetInterpolantDegree},
+     {"linear", ARKodeSetLinear},
+     {"autonomous", ARKodeSetAutonomous},
+     {"deduce_implicit_rhs", ARKodeSetDeduceImplicitRhs},
+     {"lsetup_frequency", ARKodeSetLSetupFrequency},
+     {"predictor_method", ARKodeSetPredictorMethod},
+     {"max_nonlin_iters", ARKodeSetMaxNonlinIters},
+     {"max_hnil_warns", ARKodeSetMaxHnilWarns},
+     {"interpolate_stop_time", ARKodeSetInterpolateStopTime},
+     {"max_num_constr_fails", ARKodeSetMaxNumConstrFails},
+     {"adaptivity_adjustment", ARKodeSetAdaptivityAdjustment},
+     {"small_num_efails", ARKodeSetSmallNumEFails},
+     {"max_err_test_fails", ARKodeSetMaxErrTestFails},
+     {"max_conv_fails", ARKodeSetMaxConvFails}};
+  int num_int_keys = 15;
+
+  static struct arkKeyLongPair long_pairs[] = {
+    {"max_num_steps", ARKodeSetMaxNumSteps}};
+  int num_long_keys = 1;
+
+  static struct arkKeyRealPair real_pairs[] =
+    {{"nonlin_crdown", ARKodeSetNonlinCRDown},
+     {"nonlin_rdiv", ARKodeSetNonlinRDiv},
+     {"delta_gamma_max", ARKodeSetDeltaGammaMax},
+     {"nonlin_conv_coef", ARKodeSetNonlinConvCoef},
+     {"init_step", ARKodeSetInitStep},
+     {"min_step", ARKodeSetMinStep},
+     {"max_step", ARKodeSetMaxStep},
+     {"stop_time", ARKodeSetStopTime},
+     {"fixed_step", ARKodeSetFixedStep},
+     {"step_direction", ARKodeSetStepDirection},
+     {"cfl_fraction", ARKodeSetCFLFraction},
+     {"safety_factor", ARKodeSetSafetyFactor},
+     {"error_bias", ARKodeSetErrorBias},
+     {"max_growth", ARKodeSetMaxGrowth},
+     {"min_reduction", ARKodeSetMinReduction},
+     {"max_first_growth", ARKodeSetMaxFirstGrowth},
+     {"max_efail_growth", ARKodeSetMaxEFailGrowth},
+     {"max_cfail_growth", ARKodeSetMaxCFailGrowth}};
+  int num_real_keys = 18;
+
+  static struct arkKeyActionPair action_pairs[] =
+    {{"nonlinear", ARKodeSetNonlinear},
+     {"clear_stop_time", ARKodeClearStopTime},
+     {"no_inactive_root_warn", ARKodeSetNoInactiveRootWarn},
+     {"reset_accumulated_error", ARKodeResetAccumulatedError}};
+  int num_action_keys = 4;
+
+  int i, j, retval;
+  for (i = 1; i < argc; i++)
+  {
+    sunbooleantype arg_used = SUNFALSE;
+
+    /* skip command-line arguments that do not begin with "arkode." */
+    static const char* prefix = "arkode.";
```

Comment:
Just repeating my message on Slack for posterity

What about this format: `<arkid>.<option>` where `arkid` defaults to `arkode.` if `NULL`. I think this has two benefits:
- Users can better control nesting and prefix of option
- If a user wants to swap integrators without rewriting each option, they can use an empty or consistent prefix among the command line parsing functions.